version: '3.8'
services:
  db:
    image: postgres:14
    container_name: postgres_db_certificados
    restart: always
    env_file:
      - .env
    environment:
      POSTGRES_USER: ${DATABASE_USER}
      POSTGRES_PASSWORD: ${DATABASE_PASSWORD}
      POSTGRES_DB: ${DATABASE_NAME}
    ports:
      - "5432:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data
    networks:
      - app-network
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U ${DATABASE_USER} -d ${DATABASE_NAME}"]
      interval: 5s
      timeout: 5s
      retries: 5

  web:
    build:
      context: .
      dockerfile: docker/Dockerfile
    container_name: certificados_web
    restart: always
    volumes:
      - ./:/app
      - ./data:/app/data
    ports:
      - "5000:5000"
    depends_on:
      db:
        condition: service_healthy
    env_file:
      - .env
    environment:
      DB_USER: ${DATABASE_USER}
      DB_PASSWORD: ${DATABASE_PASSWORD}
      DB_HOST: db
      DB_PORT: 5432
      DB_NAME: ${DATABASE_NAME}
      DEBUG: ${DEBUG}
    networks:
      - app-network

networks:
  app-network:
    driver: bridge

volumes:
  postgres_data: